DQMServices/Core/src/DQMStore.cc

```diff
@@ -405,14 +405,14 @@ void DQMStore::mergeAndResetMEsRunSummaryCache(uint32_t run,
 	{
 	  if(me->getTH1()->CanExtendAllAxes() && i->getTH1()->CanExtendAllAxes()) {
 	    TList list;
-	    list.Add(i->getTH1());
+	    list.Add(i->getTH1().operator->().operator->());
```

Comment:
Sorry, I'm confused :-(

The result of `i->getTH1().operator->().operator->()` is just a plain pointer, so its value cannot go away - the object is owned by the `DQMStore` in the first place.

Do you mean that the lock could be released before the call to `Add()` ?
Yes, that definitely happens, but it's not an issue here, because `mergeAndResetMEsRunSummaryCache()` should only be called in the end run transition, when the plots are not being updated.

In fact, this is where the `DQMStore` could use its "friendship" with the `MonitorElement` and access the histograms directly, without the locking.